controllers/hyperconverged/hyperconverged_controller_test.go

```diff
@@ -670,7 +670,7 @@ var _ = Describe("HyperconvergedController", func() {
 				).To(BeNil())
 
 				Expect(foundResource.Status.RelatedObjects).ToNot(BeNil())
-				Expect(len(foundResource.Status.RelatedObjects)).Should(Equal(20))
+				Expect(len(foundResource.Status.RelatedObjects)).Should(Equal(17))
```

Comment:
Can you please also add a couple of unit tests tor the removal of leftovers during upgrades like:
"should remove ConfigMap v2v-vmware upgrading from <= 1.6.0" on line 1975?

Reply:
Added, but I think I do not understand everything here correctly. The test helped me to find correct api versions (which is good) but not it is failing on line 2279:
```
					for _, objRef := range toBeRemovedRelatedObjects {
						Expect(foundResource.Status.RelatedObjects).ToNot(ContainElement(objRef))
					}
```

Reply:
I fear that we have something unexpected in the deserialization of groupVersionKind

Reply:
does it mean a problem with the PR or maybe a problem with the HCO logic

Reply:
switching from:
```
if obj.GroupVersionKind() == gvk ...
```
to 
```
apiVersion, kind := gvk.ToAPIVersionAndKind()
if obj.APIVersion == apiVersion && obj.Kind == kind ...
```
in removeRelatedObject in HCO fixed it, but honestly the root cause is still quite confusing for me.